typer/_typing.py

```diff
@@ -1,630 +0,0 @@
-        converted = tuple(
-            ForwardRef(arg)
-            if isinstance(arg, str) and isinstance(tp, TypingGenericAlias)
-            else convert_generics(arg)
-            for arg in args
-        )
-
-        if converted == args:
-            return tp
-        elif isinstance(tp, TypingGenericAlias):
-            return TypingGenericAlias(origin, converted)
-        elif isinstance(tp, TypesUnionType):
-            # recreate types.UnionType (PEP604, Python >= 3.10)
-            return _UnionGenericAlias(origin, converted)
-        else:
-            try:
-                setattr(tp, "__args__", converted)  # noqa: B010
-            except AttributeError:
-                pass
-            return tp
-
-
-if sys.version_info < (3, 10):
-
-    def is_union(tp: Optional[Type[Any]]) -> bool:
-        return tp is Union
-
-    WithArgsTypes = (TypingGenericAlias,)
-
-else:
-    import types
-    import typing
-
-    def is_union(tp: Optional[Type[Any]]) -> bool:
-        return tp is Union or tp is types.UnionType  # noqa: E721
-
-    WithArgsTypes = (typing._GenericAlias, types.GenericAlias, types.UnionType)
-
-
-if sys.version_info < (3, 9):
-    StrPath = Union[str, PathLike]
-else:
-    StrPath = Union[str, PathLike]
-    # TODO: Once we switch to Cython 3 to handle generics properly
-    #  (https://github.com/cython/cython/issues/2753), use following lines instead
-    #  of the one above
-    # # os.PathLike only becomes subscriptable from Python 3.9 onwards
-    # StrPath = Union[str, PathLike[str]]
-
-
-if TYPE_CHECKING:
-    # Only in Pydantic
-    # from .fields import ModelField
-
-    TupleGenerator = Generator[Tuple[str, Any], None, None]
-    DictStrAny = Dict[str, Any]
-    DictAny = Dict[Any, Any]
-    SetStr = Set[str]
-    ListStr = List[str]
-    IntStr = Union[int, str]
-    AbstractSetIntStr = AbstractSet[IntStr]
-    DictIntStrAny = Dict[IntStr, Any]
-    MappingIntStrAny = Mapping[IntStr, Any]
-    CallableGenerator = Generator[AnyCallable, None, None]
-    ReprArgs = Sequence[Tuple[Optional[str], Any]]
-    AnyClassMethod = classmethod[Any]
-
-__all__ = (
-    "ForwardRef",
-    "Callable",
-    "AnyCallable",
-    "NoArgAnyCallable",
-    "NoneType",
-    "is_none_type",
-    "display_as_type",
-    "resolve_annotations",
-    "is_callable_type",
-    "is_literal_type",
-    "all_literal_values",
-    "is_namedtuple",
-    "is_typeddict",
-    "is_new_type",
-    "new_type_supertype",
-    "is_classvar",
-    "update_field_forward_refs",
-    "update_model_forward_refs",
-    "TupleGenerator",
-    "DictStrAny",
-    "DictAny",
-    "SetStr",
-    "ListStr",
-    "IntStr",
-    "AbstractSetIntStr",
-    "DictIntStrAny",
-    "CallableGenerator",
-    "ReprArgs",
-    "AnyClassMethod",
-    "CallableGenerator",
-    "WithArgsTypes",
-    "get_args",
-    "get_origin",
-    "get_sub_types",
-    "typing_base",
-    "get_all_type_hints",
-    "is_union",
```

Comment:
Note that the unmerged PR https://github.com/tiangolo/typer/pull/548 uses `is_union` (as well as  `get_args` and `get_origin`) so either way I'd recommend merging that PR before we handle this one.